### PR TITLE
Adds getPendingFederation tests

### PR DESCRIPTION
**Pull request by jeremy-then**
Adds getPendingFederation tests

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 1]
```diff
@@ -892,6 +892,60 @@ void testSaveOldFederation(
         );
     }
 
+    @Test
+    void savePendingFederation_postWasabi_shouldBeSavedInStorageWithFormatVersion() {
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:896]:
```suggestion
    void savePendingFederation_whenWasabiIsActive_shouldBeSavedInStorageWithFormatVersion() {
```

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 2]
```diff
@@ -892,6 +892,60 @@ void testSaveOldFederation(
...
+
+        // Arrange
+
+        ActivationConfig.ForBlock wasabiActivation = ActivationConfigsForTest.wasabi100().forBlock(0L);
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:900]:
```suggestion
        ActivationConfig.ForBlock wasabiActivations = ActivationConfigsForTest.wasabi100().forBlock(0L);
```

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 3]
```diff
@@ -892,6 +892,60 @@ void testSaveOldFederation(
...
+
+        // Assert
+
+        assertEquals(STANDARD_MULTISIG_FEDERATION.getFormatVersion(), formatVersion);
+        assertEquals(expectedPendingFederation, actualPendingFederationInStorage);
+
+    }
+
+    @Test
+    void savePendingFederation_preWasabi_shouldBeSavedInStorageSerializedFromBtcKeysOnly() {
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:923]:
```suggestion
    void savePendingFederation_preWasabi_shouldBeSavedInStorageSerializedFromBtcKeysOnly() {
```

This test should be before the postWasabi

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 4]
```diff
@@ -836,6 +836,60 @@ void testSaveOldFederation(
...
+
+        // Arrange
+
+        ActivationConfig.ForBlock orchidActivation = ActivationConfigsForTest.orchid().forBlock(0L);
+        PendingFederation expectedPendingFederation = new PendingFederationBuilder().build();
+        StorageAccessor storageAccessor = new InMemoryStorage();
+
+        // Act
```
Comment (wilmerrootstock) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:875]: In the Act part, should be the actual result of the method we are testing. We could avoid adding the arrange part in this section. E.g.:

// Act
`federationStorageProvider.save(networkParameters, orchidActivation);`

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 1 above)
Comment (wilmerrootstock) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:840]: @julia-zack recommended me a while ago the following format for test names, and I consider it makes easier to understand what the test does:

`methodName_when<testCase>_shouldReturn/shouldSave`

In this case, we might put something like:

`save_whenWasabiIsActive_shouldSavePendingFederationInStorageWithFormatVersion()
`

WDYT, guys?
C.C.: @jeremy-then @marcos-iov @nathanieliov @apancorb @julia-zack

Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:840]: This was voted for in my polly. Let's take the conversation there if needed.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 2 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 3 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Already discussed in the poll.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 4 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 5]
```diff
@@ -836,6 +836,60 @@ void testSaveOldFederation(
         );
     }
 
+    @Test
+    void savePendingFederation_preWasabi_shouldBeSavedInStorageSerializedFromBtcKeysOnly() {
+
+        // Arrange
+
+        ActivationConfig.ForBlock orchidActivation = ActivationConfigsForTest.orchid().forBlock(0L);
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:844]:
```suggestion
        ActivationConfig.ForBlock orchidActivations = ActivationConfigsForTest.orchid().forBlock(0L);
```

**Review by nathanieliov — APPROVED** [review]

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**